### DISCUSSION TITLE
knot resolver along cloudlared

### DISCUSSION THREAD

**Discussion by malikshi** [trinib/Adguard-Wireguard-Unbound-Cloudflare#47] [Help]
i saw the latest wiki about install knot resolver
```conf
-- are to be obtained from IP address 127.0.0.1 port 6053
-- This disables DNSSEC validation!
policy.add(policy.suffix(
    policy.STUB('127.0.0.1@6053'),
    {todname('1.168.192.in-addr.arpa')}))

-- 127.0.0.1 port 6053 for proxy (dnscrypt-proxy or cloudflared)
```

i set my cloudflared with port 5053 and cloudlfared from this repo also use 5053 but in wiki knot using 6053 little bit make me confused about knot

**Comment by trinib** [comment]
> i saw the latest wiki about install knot resolver
> 
> ```
> -- are to be obtained from IP address 127.0.0.1 port 6053
> -- This disables DNSSEC validation!
> policy.add(policy.suffix(
>     policy.STUB('127.0.0.1@6053'),
>     {todname('1.168.192.in-addr.arpa')}))
> 
> -- 127.0.0.1 port 6053 for proxy (dnscrypt-proxy or cloudflared)
> ```
> 
> i set my cloudflared with port 5053 and cloudlfared from this repo also use 5053 but in wiki knot using 6053 little bit make me confused about knot

@malikshi  thanks I missed that

@jo20201 had just choose that port (6053) for DNScrypt, I think it works faster for him at his location instead what I recommended in DNScrypt wiki(5353). Port 5053 is used/recommended by Cloudflare.

**Reply by malikshi** [reply]
i think put it another mention about this outside config much better so users can read and makes changes according to their config.. mostly users just copy pasta.

**Reply by trinib** [reply]
> i think put it another mention about this outside config much better so users can read and makes changes according to their config.. mostly users just copy pasta.

I will set it to 5353 to the same i have in dnscrypt wiki for less confusion

you can test  'The Registered Ports (1024-49151) – which can be used by applications, specific services, and users'

But port like 5353, 5335, etc are recommeded by dns softwares

**Reply by malikshi** [reply]
> But port like 5353, 5335, etc are recommeded by dns softwares

yes. recommendation from dns softwares itself much better on mostly cases i think.

**Comment by malikshi** [comment]
found typo in this line 17
`        'hints > iterate'',      -- Allow loading /etc/hosts or custom root hints`

**Comment by trinib** [comment]
@malikshi  fixed wiki

![image](https://user-images.githubusercontent.com/18756975/172029054-d0f1e0d9-3163-40f1-9f27-9dc01dad5554.png)
![image](https://user-images.githubusercontent.com/18756975/172029099-19ea8df5-77dd-4e72-9f5d-42e472eb5b08.png)

**Reply by malikshi** [reply]
thank you for your hard work. fast ever!

**Reply by trinib** [reply]
> i think put it another mention about this outside config much better so users can read and makes changes according to their config.. mostly users just copy pasta.

Yea your right I should write this outside config file

**Reply by trinib** [reply]
@malikshi This should help out a lot right ?
![image](https://user-images.githubusercontent.com/18756975/172387158-9e297126-ecd7-4e70-ac6c-2cfc414d566e.png)

**Reply by malikshi** [reply]
yes it is. thanks you very much.
sir i want to ask how much resources for AGH in VPS? i have vps 2GB RAM with 2 vCPU. i planning to use agh filtering to work along with wireguard/openvpn or kind of proxy i use. currently cloudflard+knot-resolver, but i don't know to set client dns from agh. do you have reference for it?

**Reply by trinib** [reply]
> yes it is. thanks you very much. sir i want to ask how much resources for AGH in VPS? i have vps 2GB RAM with 2 vCPU. i planning to use agh filtering to work along with wireguard/openvpn or kind of proxy i use. currently cloudflard+knot-resolver, but i don't know to set client dns from agh. do you have reference for it?

2gb is good.. what ever port dns resolvers listen on from local address (127.0.0.1),  need to be in adguard dns resolvers.
